src/unix/async.c

```diff
@@ -58,7 +58,11 @@ int uv_async_init(uv_loop_t* loop, uv_async_t* handle, uv_async_cb async_cb) {
 
 
 int uv_async_send(uv_async_t* handle) {
-  if (uv__async_make_pending(&handle->pending) == 0)
+  /* Do a cheap read first. */
+  if (ACCESS_ONCE(int, handle->pending) != 0)
```

Comment:
It does on one benchmark (under really high contention) and doesn't noticeably affect the other benchmarks. It reduces the "async_pummel_8" bench by about half on my 4-core x86 proc. I'd like to remove it too. :) It reminds me of double-checked locking (which is notoriously easy to get wrong). I don't know enough about the low-level details to know how this will behave on OOE processors (arm, power, etc.) or if that's even a concern.